### PR TITLE
Feature/waiting for scm

### PR DESCRIPTION
**Pull request by msuewer**
Problem: If scm manager is noch reachable, an error is thrown that is not handled. Because of that, the smeagol page is not shown but the error with an error message that does not lead to the actual problem (that the scm manager is not reachable).
Solution:
1. Handling error and show user an error message that shows that scm manager is not reachable
2. Healthy check for scm at start of smeagol

### CODE REVIEW

File: src/main/java/com/cloudogu/wiki/WikiDispatcherServlet.java [hunk 1]
```diff
@@ -53,8 +53,15 @@ protected void service(HttpServletRequest req, HttpServletResponse resp) throws
         String repositoryId = getRepositoryId(req);
         String branchName = getBranchName(req);
 
+
         if (Strings.isNullOrEmpty(repositoryId)) {
-            renderOverview(req, resp);
+            try {
+                renderOverview(req, resp);
+            }
+            catch(RuntimeException ex){
```
Comment (sdorra) [inline src/main/java/com/cloudogu/wiki/WikiDispatcherServlet.java]: Do not catch RuntimeException, please catch only the exception which is responsible for the connection problem.

File: ces-startup.sh [hunk 2]
```diff
@@ -9,6 +9,16 @@ export SMEAGOL_CAS_URL="https://${FQDN}/cas"
...
 export PLANTUML_URL="https://${FQDN}/plantuml/png"
 
+
+# wait until scm passes all health checks
+echo "wait until scm passes all health checks"
+if ! doguctl healthy --wait --timeout 120 scm; then
+  echo "timeout reached by waiting of scm to get healthy"
+  exit 1
+fi
+sleep 10
```
Comment (sdorra) [inline ces-startup.sh]: Do we really need another sleep for 10 seconds?

Comment (msuewer) [reply ces-startup.sh]: Testing shows it does not change anything, so I removed it now.

File: src/main/java/com/cloudogu/wiki/scmm/ScmManager.java [hunk 3]
```diff
@@ -51,7 +52,7 @@ private ScmManager() {
 
             return wikis;
         } catch (UnirestException ex) {
```
Comment (cesmarvin) [inline src/main/java/com/cloudogu/wiki/scmm/ScmManager.java]: CRITICAL Either log or rethrow this exception. [rule](https://ecosystem.cloudogu.com/sonar/coding_rules#rule_key=squid%3AS1166)

**Comment by cesmarvin** [comment]
SonarQube analysis reported 1 issue
* ![CRITICAL][CRITICAL] 1 critical

Watch the comments in this conversation to review them.

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'

**Pull request merged**